docs/design-docs/overview.md

```diff
@@ -0,0 +1,144 @@
+    5. Prepare output path.
+    6. Start SSH service and check inter-connections.
+5.	Sb runner loops all benchmarks and all modes in each benchmark. For each mode in each benchmark, sb runner calls sb executor inside Docker container on corresponding compute nodes to start an execution.
+6.	Sb executor parses the config object, start one benchmark inside Docker container.
+7.	Sb executor gets return code and results once the benchmark finished.
+8.	Sb executor sends the return code and results back to sb runner. Sb runner checks return code and moves to the next execution.
+9.	Once finished, sb runner reduces results on all compute nodes, save log and results files, then summarize running results to sb cli.
+10.	Sb cli returns what it gets to user.
+
+## Components
+
+### SB CLI
+
+SB cli is a command line interface for end users to use SuperBench tool and run benchmarks. The cli should provide a set of commands and corresponding help information to users. Details will be discussed in [interfaces section](#interfaces).
+
+### SB Runner
+
+SB runner is the component to configure environments, prepare context, run benchmarks, and collect log and results on all compute nodes. It will control the running logic, including when to start, which node or a set of nodes to run, whether a barrier is needed, etc. Sb runner will either communicate with host through SSH to configure running environments and prepare context, or communicate with sb executor inside Docker container to execute benchmarks and collect log and results on each node.
+
+Here’re the details about running context and output structure for sb runner.
+
+**Context**
+
+```
+/path/to/working/directory
+├── outputs				# log directory
+│   └── 2020-12-18_20-43-42		# for each run, name in %Y-%m-%d_%H-%M-%S format
+│       ├── config.yaml		# dumped config object for internal use
+│       ├── ssh_config		# ~/.ssh/config for all hosts
+│       ├── id_rsa			# generated ssh private key for each run
+│       └── id_rsa.pub		# generated ssh public key for each run
+├── superbench				# local superbench code
+├── requirements.txt
+└── setup.py
+```
+
+**Output**
+
+```
+/path/to/working/directory
+└── outputs				# log directory
+    └── 2020-12-18_20-43-42		# for each run, name in %Y-%m-%d_%H-%M-%S format
+        ├── sb-run.log		# sb runner log
+        ├── node0			# directory for each node
+        │   ├── sb-exec.log		# sb executor log for each node
+        │   └── benchmarks		# results directory for benchmarks
+        │       ├── nccl		# nccl result
+        │       ├── pytorch_models	# pytorch models result
+        │       └── ...		# others
+        └── node...
+```
+
+### SB Executor
+
+SB executor is the component to run benchmarks inside Docker container. It will execute each benchmark actually and handle all pre- and post-processing, including health check, result validation, result processing, etc.
```

Comment:
`SuperBenchExecutor` is the component to run benchmarks inside Docker container. It will execute each benchmark and handle all pre- and post-processing, including health check, result validation, result processing, etc.